postgresql/provider.go

```diff
@@ -170,12 +212,28 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	versionStr := d.Get("expected_version").(string)
 	version, _ := semver.ParseTolerant(versionStr)
 
+	host := d.Get("host").(string)
+	port := d.Get("port").(int)
+	username := d.Get("username").(string)
+
+	var password string
+	if d.Get("aws_rds_iam_auth").(bool) {
+		profile := d.Get("aws_rds_iam_profile").(string)
+		var err error
+		password, err = getRDSAuthToken(profile, username, host, port)
```

Comment:
Just to note, this password has a limited TTL, I think 15m.

This approach of setting the pwd here should be OK if the terraform execution is shorter.

But ideally we should generate the password every time we create a new DB connection. Maybe here: https://github.com/cyrilgdn/terraform-provider-postgresql/blob/master/postgresql/config.go#L261

To wire this, maybe we could add a callback to the Config object, something like `onConnect` or so. The callback would be called before connect.

Other drivers have hooks for this purpose: https://pkg.go.dev/github.com/jackc/pgx/v4/stdlib#OptionBeforeConnect